crates/astria-sequencer/src/transaction/unsigned.rs

```diff
@@ -0,0 +1,122 @@
+use astria_proto::sequencer::v1::{
+    unsigned_transaction::Value::AccountsTransaction as ProtoAccountsTransaction,
+    UnsignedTransaction as ProtoUnsignedTransaction,
+};
+use prost::Message as _;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+use crate::{
+    accounts::transaction::Transaction as AccountsTransaction,
+    crypto::SigningKey,
+    hash,
+    transaction::signed::Transaction as SignedTransaction,
+};
+
+/// Represents an unsigned sequencer chain transaction.
+/// This type wraps all the different module-specific transactions.
+/// If a new transaction type is added, it should be added to this enum.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+pub(crate) enum Transaction {
+    AccountsTransaction(AccountsTransaction),
+}
+
+impl Transaction {
+    /// Attempts to encode the unsigned transaction into bytes.
+    #[must_use]
+    pub(crate) fn to_vec(&self) -> Vec<u8> {
+        match &self {
+            Transaction::AccountsTransaction(tx) => ProtoUnsignedTransaction {
+                value: Some(ProtoAccountsTransaction(tx.to_proto())),
+            },
+        }
+        .encode_length_delimited_to_vec()
+    }
+
+    /// Signs the transaction with the given keypair.
+    #[allow(dead_code)]
```

Comment:
Not a fan of allowing dead code for the sake of running tests....

Can you just make this a free-standing fn inside the test?

Reply:
this is used in multiple unit tests so I didn't want to copy-paste, as well it's used here in the client crate: https://github.com/astriaorg/astria/pull/129 so it's no longer dead code

Reply:
ok